### ISSUE TITLE
Crash due to Holes in Function Types

### ISSUE THREAD

**Issue by christetreault** [ucsd-progsys/liquidhaskell#479]
I'm getting the following benchmark failure on the master branch:

```
    hscolour
      HsColour.hs:                                  FAIL (1.75s)
        Wrong exit code
        expected: ExitSuccess
         but got: ExitFailure 2
```

It seems that the hscolour benchmark suite is in a git submodule, which may or may not have anything to do with it. All other source files in this subfolder pass. 

`XMonad.hs` is also in a git submodule, and if this is somehow the cause of this issue, it might fail also. However, nothing in XMonad folder got tested for whatever reason (probably an issue in its own right)

**Comment by spinda** [comment]
Can you try `git submodule update --recursive`?

I supect it's not the submodules, though, but a bug that's been allowed to creep in since the `HsColour` benchmark isn't enabled on Travis.

**Comment by christetreault** [comment]
No luck, same result.

**Comment by christetreault** [comment]
And while I'm looking at it, it seems that xmonad isn’t in the benchmarks suite. Is this intentional?

**Comment by ranjitjhala** [comment]
These should NOT be sub modules ...  Perhaps the result of stale .git directories -- can we remove them?

**Comment by ranjitjhala** [comment]
It has nothing to do with submodule, but is the same problem as (the new) `tests/pos/hole-fun.hs` 
assigning @gridaphobe

**Comment by gridaphobe** [comment]
Fixed by #492
